tsdb/mapper.go

```diff
@@ -93,8 +160,18 @@ func (lm *SelectMapper) timeDirection() Direction {
 	return Forward
 }
 
-// Open opens the local mapper.
+// Open opens the Mapper for a SELECT Statement
 func (lm *SelectMapper) Open() error {
+	// If in aggregate mode, initialize the map-and-reduce functions. Both local and
+	// remote mappers need this.
+	if !lm.rawMode {
```

Comment:
This was moved up here because it now needs to be done before the remote mapper is opened.